pkg/storage/secret/metadata/outbox_store_test.go

```diff
@@ -0,0 +1,225 @@
+		Type:            contracts.CreateSecretOutboxMessage,
+		Name:            "s-1",
+		Namespace:       "n-1",
+		EncryptedSecret: secretv0alpha1.NewExposedSecureValue("value"),
+		KeeperName:      contracts.DefaultSQLKeeper,
+		ExternalID:      nil,
+	}
+
+	outboxMessage2 := contracts.OutboxMessage{
+		MessageID:       "message_id_2",
+		Type:            contracts.CreateSecretOutboxMessage,
+		Name:            "s-1",
+		Namespace:       "n-1",
+		EncryptedSecret: secretv0alpha1.NewExposedSecureValue("value"),
+		KeeperName:      contracts.DefaultSQLKeeper,
+		ExternalID:      nil,
+	}
+
+	model.Append("message_id_1", appendOutboxMessage)
+
+	require.Equal(t, []contracts.OutboxMessage{outboxMessage1}, model.ReceiveN(10))
+
+	model.Append("message_id_2", appendOutboxMessage)
+
+	require.Equal(t, []contracts.OutboxMessage{outboxMessage1, outboxMessage2}, model.ReceiveN(10))
+
+	model.Delete(outboxMessage1.MessageID)
+
+	require.Equal(t, []contracts.OutboxMessage{outboxMessage2}, model.ReceiveN(5))
+
+	model.Delete(outboxMessage2.MessageID)
+
+	require.Empty(t, model.ReceiveN(1))
+}
+
+func TestOutboxStore(t *testing.T) {
+	testDB := sqlstore.NewTestStore(t)
+	require.NoError(t, migrator.MigrateSecretSQL(testDB.GetEngine(), nil))
+
+	ctx := context.Background()
+
+	outbox := ProvideOutboxQueue(testDB)
+
+	appendOutboxMessage := contracts.AppendOutboxMessage{
+		Type:            contracts.CreateSecretOutboxMessage,
+		Name:            "s-1",
+		Namespace:       "n-1",
+		EncryptedSecret: secretv0alpha1.NewExposedSecureValue("value"),
+		KeeperName:      contracts.DefaultSQLKeeper,
+		ExternalID:      nil,
+	}
+
+	messages, err := outbox.ReceiveN(ctx, 10)
+	require.NoError(t, err)
+	require.Empty(t, messages)
+
+	messageID1, err := outbox.Append(ctx, appendOutboxMessage)
+	require.NoError(t, err)
+
+	for range 2 {
+		messages, err = outbox.ReceiveN(ctx, 10)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(messages))
+		require.Equal(t, messageID1, messages[0].MessageID)
+	}
+
+	messageID2, err := outbox.Append(ctx, appendOutboxMessage)
+	require.NoError(t, err)
+
+	messages, err = outbox.ReceiveN(ctx, 3)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(messages))
+	require.Equal(t, messageID1, messages[0].MessageID)
+	require.Equal(t, messageID2, messages[1].MessageID)
+
+	require.NoError(t, outbox.Delete(ctx, messageID1))
+	messages, err = outbox.ReceiveN(ctx, 10)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(messages))
+	require.Equal(t, messageID2, messages[0].MessageID)
+
+	require.NoError(t, outbox.Delete(ctx, messageID2))
+
+	messages, err = outbox.ReceiveN(ctx, 100)
+	require.NoError(t, err)
+	require.Empty(t, messages)
+}
+
+func TestOutboxStoreProperty(t *testing.T) {
+	seed := time.Now().UnixMicro()
+	rng := rand.New(rand.NewSource(seed))
+
+	defer func() {
+		if err := recover(); err != nil || t.Failed() {
+			fmt.Printf("TestOutboxStoreProperty: err=%+v\n\nSEED=%+v", err, seed)
+			t.FailNow()
+		}
+	}()
+
+	for range 10 {
```

Comment:
How were the numbers on the `for range` (and the one below) decided? Could you document, even if it was arbitrary?

Reply:
Arbitrary based on the time the test takes to run. Documented